shard/committee/assignment.go

```diff
@@ -381,6 +382,15 @@ func eposStakedCommittee(
 	}
 
 	if len(completedEPoSRound.AuctionWinners) == 0 {
+		instance := shard.Schedule.InstanceForEpoch(epoch)
+		preInstance := shard.Schedule.InstanceForEpoch(new(big.Int).Sub(epoch, big.NewInt(1)))
+		isTestnet := nodeconfig.GetDefaultConfig().GetNetworkType() == nodeconfig.Testnet
+		isShardReduction := preInstance.NumShards() != instance.NumShards()
+		// If the shard-reduction happens, we cannot use the old committee.
+		if isTestnet && isShardReduction {
```

Comment:
At this point, the `shardState` has the correct number of shards so it can be safely returned. We are not doing the committee calculation here; it's already done at line 365 based on `s.NumShards()`, 2 in this case.

The purpose of this block, before this PR, was to ensure at least one external validator is always elected even if it is not signing (`len(completedEPoSRound.AuctionWinners) == 0`). This was done by reusing the prior epoch's shard state.

With the testnet shard reduction, however, the prior epoch's shard state is no longer usable (since it has 2 more shards). Therefore, we cannot return the old shard state, it must be the new one even if it does not have any external validators. 

The `isTestnet && isShardReduction` check is put inside the empty committee case to prevent doing that check each time, even when there is an elected external validator.